G/Git/build_tarballs.jl

```diff
@@ -75,8 +75,17 @@ dependencies = [
 
 # Install first for win32, then win64.  This will accumulate files into `products` and also wrappers into the JLL package.
 non_reg_ARGS = filter(arg -> arg != "--register", ARGS)
-build_tarballs(non_reg_ARGS, name, version, sources_w32, script_win, [Windows(:i686)], products, [])
-build_tarballs(non_reg_ARGS, name, version, sources_w64, script_win, [Windows(:x86_64)], products, [])
 
+include("../../fancy_toys.jl")
+
+if should_build_platform("i686-w64-mingw32")
+    build_tarballs(non_reg_ARGS, name, version, sources_w32, script_win, [Windows(:i686)], products, [])
+end
+if should_build_platform("x86_64-w64-mingw32")
+    build_tarballs(non_reg_ARGS, name, version, sources_w64, script_win, [Windows(:x86_64)], products, [])
+end
 # Then for everything else.  This is the only one that we try to register, and this is the step that will open a PR against General
-build_tarballs(ARGS, name, version, sources_unix, script_unix, filter(p -> !isa(p, Windows), supported_platforms()), products, dependencies)
+platforms = filter!(p -> !isa(p, Windows), supported_platforms())
+if any(should_build_platform.(triplet.(platforms)))
+    build_tarballs(ARGS, name, version, sources_unix, script_unix, platforms, products, dependencies)
```

Comment:
I think you need to `filter!(p -> should_build_platform(triplet(p)), platforms)` then `if !isempty(platforms)` instead of `if any()`, since this will build all the non-windows platforms if we ask for a single non-windows platform.

Reply:
The two Windows build jobs in https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=388&view=results would have had this problem, no?

Reply:
```julia
julia> ARGS = ["i686-w64-mingw32"]
1-element Array{String,1}:
 "i686-w64-mingw32"

julia> platforms = filter!(p -> !isa(p, Windows), supported_platforms()) 
11-element Array{Platform,1}:
 Linux(:i686, libc=:glibc)                    
 Linux(:x86_64, libc=:glibc)                  
 Linux(:aarch64, libc=:glibc)                 
 Linux(:armv7l, libc=:glibc, call_abi=:eabihf)
 Linux(:powerpc64le, libc=:glibc)             
 Linux(:i686, libc=:musl)                     
 Linux(:x86_64, libc=:musl)                   
 Linux(:aarch64, libc=:musl)                  
 Linux(:armv7l, libc=:musl, call_abi=:eabihf) 
 MacOS(:x86_64)                               
 FreeBSD(:x86_64)                             

julia> any(should_build_platform.(triplet.(platforms)))
false

julia> should_build_platform("i686-w64-mingw32")
true

julia> should_build_platform("x86_64-w64-mingw32")
false
```

Reply:
I mean that if you ask for a single **non-windows** platform, you will build many non-windows platforms:

```
ARGS = ["x86_64-linux-gnu"]
platforms = filter!(p -> !isa(p, Windows), supported_platforms())
any(should_build_platform.(triplet.(platforms)))
```

That returns `true`, then we pass the full `platforms` into `build_tarballs()`, which means that we will build for more than just `x86_64-linux-gnu`.  :)

Reply:
Aaaaah, sorry, now I see what you mean! :grin: Done!